LibationWinForms/GridEntry.cs

```diff
@@ -70,8 +76,48 @@ private void PictureStorage_PictureCached(object sender, FileManager.PictureCach
 			}
 		}
 
-		#region Data Source properties
+		#region detect changes to the model and update the view
+
+		/// <summary>
+		/// This event handler receives notifications from the model that it has changed.
+		/// Save to the database and notify the view that it's changed.
+		/// </summary>
+		private void UserDefinedItem_ItemChanged(object sender, string itemName)
+		{
+			var udi = sender as UserDefinedItem;
+
+			if (udi.Book.AudibleProductId != LibraryBook.Book.AudibleProductId)
```

Comment:
These are fine as is, but in case it helps: LibraryBook.Book is aliased above as just Book since it's used so frequently.